server.go

```diff
@@ -648,21 +721,18 @@ func (s *server) findPeer(peerKey *btcec.PublicKey) (*peer, error) {
 // cleans up all resources allocated to the peer, notifies relevant sub-systems
 // of its demise, and finally handles re-connecting to the peer if it's
 // persistent.
-//
 // NOTE: This MUST be launched as a goroutine.
 func (s *server) peerTerminationWatcher(p *peer) {
+	defer p.wg.Done()
```

Comment:
Since its is a bit confusing that this is tracked by the peer's waitgroup, could you add to the godoc the comment explaining that this is the case?